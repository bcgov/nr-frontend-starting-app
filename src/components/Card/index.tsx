import React from 'react';

interface CardProps {
  title: string,
  text: string,
  children?: React.ReactNode,
  // data-test?: string
}

const Card = ({ title, text, children }: CardProps) => (
  <div className="card">
    <h4 className="card-header">{ title }</h4>
    <div className="card-body">
      <p className="card-text">{ text }</p>
    </div>
    <div className="card-footer">
      {children}
    </div>
  </div>
);

export default Card;
