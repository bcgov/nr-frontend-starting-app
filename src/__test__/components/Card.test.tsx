/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Card from '../../components/Card';

it('renders correctly', () => {
  const tree = renderer
    .create(<Card title="Test card" text="This is just a test text." />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
