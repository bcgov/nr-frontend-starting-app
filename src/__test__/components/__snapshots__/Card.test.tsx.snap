// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <h4
    className="card-header"
  >
    Test card
  </h4>
  <div
    className="card-body"
  >
    <p
      className="card-text"
    >
      This is just a test text.
    </p>
  </div>
  <div
    className="card-footer"
  />
</div>
`;
